torch/distributed/_tensor/placement_types.py

```diff
@@ -102,15 +167,27 @@ def _local_shard_size_on_dim(
         assert (
             size_on_dim >= num_chunks
         ), f"Size to be sharded on dim {self.dim} must be at least as large as the number of devices in that dimension {num_chunks}"
-        split_size, pad_idx = divmod(size_on_dim, num_chunks)
-        local_shard_size = (
-            split_size + 1 if pad_idx != 0 and rank < pad_idx else split_size
-        )
+
+        # compute the chunk size inline with ``torch.chunk``
+        full_chunk_size = (size_on_dim + num_chunks - 1) // num_chunks
+
+        # Compute chunk size for each chunk on the dimension.
+        chunk_sizes = [
+            max(
+                min(size_on_dim, full_chunk_size * (idx + 1)) - full_chunk_size * idx,
+                0,
+            )
+            for idx in range(num_chunks)
+        ]
+        local_shard_size = chunk_sizes[rank]
+
         local_offset_on_dim = -1
         if return_offset:
+            # QQ: what would be the offset of an empty shard? -1?
```

Comment:
hmmm that's a good point... do you know how sharded tensor offset on empty shard looks like? I think we might can return the "global tensor dim size" for empty shard (representing the end of that tensor dim), but would like to see if this make sense for existing use case.

Reply:
Let me double check on this.